src/Types.hs

```diff
@@ -99,6 +99,8 @@ data Either3 a b c = Left3 a | Middle3 b | Right3 c
 
 instance (SOP.Generic a, SOP.Generic b, SOP.Generic c) => SOP.Generic (Either3 a b c)
 
+app2 :: (c -> d) -> (a -> b -> c) -> (a -> b -> d)
+app2 f g = \x y -> f $ g x y
```

Comment:
We have the base case as `form`, and only tha case of `app2 form (f x y)`, I think intoroducing form2 wouldn't be better. The read the `app2` is the following: "apply an arbitrary function to the result of ...", instead of introducing a `form2` with a comment on top of it, that this one waits for a function with two arguments functions, or even further we could introduce a typeclass to form this abstraction, but this only occurs in the `Frontend` module, I wouldn't complicate things.